### DISCUSSION TITLE
Register OpenTelemetry Before Startup.cs without Dependency Injection in dotnet 7

### DISCUSSION THREAD

**Discussion by emmayspark** [open-telemetry/opentelemetry-dotnet#4603] [Q&A] [answered]
By design, we are adding additional logs before startup. I tried to create logger and register it but logs do not show up.
(It worked when I registered in ConfigurationServices in startup.cs)
Is there any workaround to register Logging before host started?

**Comment by utpilla** [comment]
@emmayspark You could use something like this:
```csharp
// Manually create logger factory
using var loggerFactory = LoggerFactory.Create(builder => builder.AddOpenTelemetry(logging => logging.AddConsoleExporter()));

var programLogger = loggerFactory.CreateLogger("Program");
programLogger.LogInformation("Starting...");

var appBuilder = WebApplication.CreateBuilder(args);

// Register manually created logger factory as the ILoggerFactory for the app
appBuilder.Services.AddSingleton<ILoggerFactory>(loggerFactory);

var app = appBuilder.Build();
app.Run();

programLogger.LogInformation("Stopping...");
```

**Comment by bsulliva** [comment]
There is a note at the bottom of this page, https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/logs/getting-started-console/README.md, that says users should not create a new LoggerFactory.

> Certain types of applications (e.g. [ASP.NET Core](https://learn.microsoft.com/aspnet/core) and [.NET Worker](https://learn.microsoft.com/dotnet/core/extensions/workers)) have an ILogger based logging pipeline set up by default. In such apps, enabling OpenTelemetry should be done by adding OpenTelemetry as a provider to the existing logging pipeline, and users should not create a new LoggerFactory (which sets up a totally new logging pipeline). Also, obtaining ILogger instance could be done differently as well. See [Example ASP.NET Core application](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Program.cs) for an example which shows how to add OpenTelemetry to the logging pipeline already setup by the application.

**Reply by cijothomas** [reply, marked as answer]
if you want to do some logging even before the asp.net core pipeline is setup, then you need to setup own LoggerFactory. And this may be thrown away after initialization is done, when you can leverage the existing logging pipeline.
